src/server/scripts/EasternKingdoms/Stratholme/stratholme.cpp

```diff
@@ -332,10 +333,48 @@ class spell_ysida_saved_credit : public SpellScript
     }
 };
 
+enum HauntingPhantoms
+{
+    SPELL_SUMMON_SPITEFUL_PHANTOM = 16334,
+    SPELL_SUMMON_WRATH_PHANTOM    = 16335
+};
+
+class spell_stratholme_haunting_phantoms : public AuraScript
+{
+    PrepareAuraScript(spell_stratholme_haunting_phantoms);
+
+    void CalcPeriodic(AuraEffect const* /*aurEff*/, bool& isPeriodic, int32& amplitude)
+    {
+        isPeriodic = true;
+        amplitude = irand(30, 90) * IN_MILLISECONDS;
+    }
+
+    void HandleDummyTick(AuraEffect const* /*aurEff*/)
+    {
+        if (irand(0, 1))
```

Comment:
`roll_chance_i(50)` ? Is this what you meant @ccrs

Reply:
Thanks, I was wondering if ccrs meant a TC function or general/external.

Reply:
yeah